sdks/go/pkg/beam/core/util/reflectx/functions_test.go

```diff
@@ -0,0 +1,43 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reflectx
+
+import (
+	"reflect"
+	"testing"
+)
+
+func testFunction() int {
+	return 42
+}
+
+func TestXxx(t *testing.T) {
+	val := reflect.ValueOf(testFunction)
+	fi := uintptr(val.Pointer())
+	typ := val.Type()
+
+	callable := LoadFunction(fi, typ)
+
+	cv := reflect.ValueOf(callable)
+	out := cv.Call(nil)
+	if len(out) != 1 {
+		t.Errorf("got %d return values, wanted 1.", len(out))
+	}
+	// TODO: check type?
+	if out[0].Int() != 42 {
+		t.Errorf("got %d, wanted 42", out[0].Int())
+	}
```

Comment:
Nit: Instead of comparing to 42 directly, could you do something like:

```
expected := testFunction()
if out[0].Int() != expected {
	t.Errorf("got %d, wanted %d", out[0].Int(), expected)
}
```

Reply:
Good point, will change.

Reply:
It's a bit more verbose but I've liked the following style since it avoids comparing then printing the wrong things accidentally.

if got,want := out[0].Int(), testFunction(); got != want {
  t.Errorf("got %d, wanted %d", got, want)
}

Reply:
I like this style, since the comma operator helps keep it compact.

Joe made an offline comment that every usage of the unsafe package should be accompanied by a unit test. I agree with that, and I'm looking at the remaining unsafe usages in the SDK right now. I'll make this tweak in that followup PR.

Reply:
Yes, please! :)

The lack of a unit test on this function meant that it took several days to track down the issue when there was mysterious memory corruption happening in the entire beam job. A single unit test would have immediately identified this as the culprit.